From a703fc895a6eecde100482ae41c9aa4f1439710a Mon Sep 17 00:00:00 2001
From: Mohsen Dolaty <mohsen.dolaty@amd.com>
Date: Thu, 2 Dec 2021 11:18:47 -0600
Subject: [PATCH 1/1] recipes-kernel/linux: remove LCD Mux from DTS

recipes-kernel/linux: remove LCD Mux from i2c-1 DTS

Removing LCD Mux to prevent Mux driver from scanning the LCD bus
LCD will hang the bus during any read operation

signed-off-by: Mohsen Dolaty <mohsen.dolaty@amd.com>
---
 arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts   | 3 ++-
 arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts | 5 +++--
 arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts   | 3 ++-
 3 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts b/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
index 64d1779614e8..66910840f3a1 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
@@ -140,12 +140,13 @@ &i2c1 {
 	//Net name i2c10 - onyx side i2c12
 	// Connected to TCA9546A and to LCD HDR
 	status = "okay";
-
+#ifdef OPENBMC_LCD_DRIVER
 	tca9546a@70 {
 		compatible = "nxp,pca9546";
 		reg = <0x70>;
 		//<TBD> Define LCD details here
 	};
+#endif //OPENBMC_LCD_DRIVER
 };

 &i2c4 {
diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts b/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
index 2f727d74eedf..2607c9219933 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
@@ -140,12 +140,13 @@ &i2c1 {
 	//Net name i2c12 (SCM_I2C12) - LCD at 50KHz
 	// Connected to TCA9546A and to LCD HDR
 	status = "okay";
-
+#ifdef OPENBMC_LCD_DRIVER
 	i2cswitch@70 {
 		compatible = "nxp,pca9546";
 		reg = <0x70>;
 		//<TBD> Define LCD details here
 	};
+#endif //OPENBMC_LCD_DRIVER
 };

 &i2c4 {
@@ -518,7 +519,6 @@ &i2c15 {
 	// RoT
 	status = "okay";
 };
-
 #ifdef OPENBMC_I3C_ENABLE
 &i3c0 {
 	// P0 DIMM (A-F) SPD Access
@@ -670,6 +670,7 @@ spd5118_3_5: spd@55 {
 	};
 };
 #endif  // OPENBMC_I3C_ENABLE
+
 &i2c2 {
 	// P0 APML
 	status = "okay";
diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts b/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts
index cf8364c7623f..9699147cf67c 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts
@@ -138,12 +138,13 @@ &i2c0 {
 &i2c1 {
 	//Net name i2c12 - Ruby side
 	status = "okay";
-
+#ifdef OPENBMC_LCD_DRIVER
 	i2cswitch@70 {
 		compatible = "nxp,pca9548";
 		reg = <0x70>;
 		//<TBD> Define LCD details here
 	};
+#endif //OPENBMC_LCD_DRIVER
 };

 &i2c2 {
--
2.25.1

